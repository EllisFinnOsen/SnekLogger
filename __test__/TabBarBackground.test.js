// TabBarBackground.test.js
import React from "react";
import { render } from "@testing-library/react-native";
import { Text } from "react-native";
import BlurTabBarBackground, {
  useBottomTabOverflow,
} from "@/components/global/TabBarBackground.ios";

// Mock dependencies for the hook
jest.mock("@react-navigation/bottom-tabs", () => ({
  useBottomTabBarHeight: () => 20, // Return a fixed value for testing
}));

jest.mock("react-native-safe-area-context", () => ({
  useSafeAreaInsets: () => ({ bottom: 20 }), // Return a fixed bottom inset
}));

describe("BlurTabBarBackground", () => {
  it("renders correctly and matches the snapshot", () => {
    const { toJSON } = render(<BlurTabBarBackground />);
    expect(toJSON()).toMatchSnapshot();
  });
});

describe("useBottomTabOverflow", () => {
  // Create a simple test component that calls the hook and renders its value inside a <Text>.
  const HookTester = () => {
    const value = useBottomTabOverflow();
    return <Text>{value}</Text>;
  };

  it("returns 0 when tab bar height equals safe area bottom", () => {
    const { getByText } = render(<HookTester />);
    expect(getByText("0")).toBeTruthy();
  });
});
