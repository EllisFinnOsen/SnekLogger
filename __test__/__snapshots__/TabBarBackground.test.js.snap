// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlurTabBarBackground renders correctly and matches the snapshot 1`] = `
<View
  intensity={100}
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  tint="systemChromeMaterial"
/>
`;
